File: src/python/Collision_avoidance.py
import math, re

class Collision_avoidance:
  MINDIST = 1000
  
  def __init__(self, invInfo, myInfo):
    invInfo[0] = self.invID
    invInfo[1] = self.invX
    invInfo[2] = self.invY
    invInfo[3] = self.invZ
    myInfo[0] = self.myID
    myInfo[1] = self.myX
    myInfo[2] = self.myY
    myInfo[3] = self.myZ
    
  def run_ca(self):
    ##figure goUp
    if (myY < invY):
        goUp = 0
    elif (myY > invY):
        goUp = 1
    else:
        if (myID > invID):
            goUp = 1
        else:
            goUp = 0
        
    
    ## calc distance
    xSqr = (self.myX - self.invX) * (self.myX - self.invX)
    ySqr = (self.myY - self.invY) * (self.myY - self.invY)
    zSqr = (self.myZ - self.invZ) * (self.myZ - self.invZ)
    mySqr = xSqr + ySqr + zSqr
    myDistance = math.sqrt(mySqr)
    if (myDistance < MINDIST):
      self.calc_adujst(myDistance,goUp)
      return 1
    else:
      return 0
    
  def calc_adjust(self,distance,goUp):
    adjustDistance = distance/2
    if (goUp =0):
      self.myY = self.myY -adjustDistance
    elif (goUp=1):
      self.myY = self.myY + adjustDistance

File: src/version_1/collision_avoidance.cpp
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <cstring>
#include <sstream>
#include <iostream>
#include <math.h>

#include "collision_avoidance.h"

#define MINDIST 8000


void Collision_avoidance::Collision_avoidance(triCoords *myCurCoords,int myFlightNum)
{
    myCoordinates = malloc(sizeof(*myCoordinates));
    myCoordinates->x = myCurCoords.x;
    myCoordinates->y = myCurCoords.y;
    myCoordinates->y = myCurCoords.z;
    myFlightID = myFlightNum;
    
}


// check to see if we need to avoid stuff
int Collision_avoidance::compute_ca(triCoords invCurCoords, int invFlightID)
{
  int myX, myY, myZ, invX, invY, invZ, myID, invID, goUp;
  myX = myCoordinates.x;
  myY = myCoordinates.y;
  myZ = myCoordinates.z;
  invX = invCurCoords.x;
  invY = invCurCoords.y;
  invZ = invCurCoords.z;
  myID = myFlightID;
  invID = invFlightID;
  
  if (myY>invY){
    goUp = 1;
  }else if (myY<invY){
    goUp = 0;
  }else {
    if (myID > invID){
      goUp = 1;
    }else if (myID < invID){
      goUp = 0;
    }
  }
  
  float xSqr = (myX - invX) * (myX - invX);
  float ySqr = (myY - invY) * (myY - invY);
  float zSqr = (myZ - invZ) * (myZ - invZ);

  double mySqr = xSqr + ySqr + zSqr;

  double myDistance = sqrt(mySqr);
  if (abs(myDistance) > MINDIST){
    return 0;
  } else if (abs(myDistance)<=MINDIST){
    calc_adjust(myDistance/2, goUp);
    return 1;
  }
}


// adjust collision course based on height only
void Collision_avoidance::calc_adjust(int dist, int goUp)
{
  int newY, adjustBy;
  adjsutBy = MINDIST-dist;
  adjustBy = adjustBy/2;
  if (goUp = 0){
    while(newY-adjustBy <= 1000){
      adjustBy = adjustBy/2;
      if (adjustBy <1) {
        adjusBy = 0;
      }
    }
    newY = myCoordinates.y-adjustBy;
    myCoordinates.y = newY;
  }else if (goUp = 1){
    newY = myCoordinates.y+adjustBy;
    myCoordinates.y = newY;
  }
}

triCoords Collision_avoidance::get_my_coords()
{
    return myCoordinates;
}


